recipes/iqtree/meta.yaml

```diff
@@ -1,40 +1,43 @@
 {% set name = "IQTREE" %}
 {% set version = "2.3.5" %}
-{% set sha256 = "8e323e0b7c46e97901d3500f11e810703e0e5d25848188047eca9602d03fa6b1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version | replace("-", "_") }}
 
 build:
-  number: 1
+  number: 2
   run_exports:
     - {{ pin_subpackage('iqtree', max_pin="x") }}
 
 source:
-  - url: https://github.com/iqtree/iqtree2/archive/refs/tags/v{{ version }}.tar.gz
-    sha256: {{ sha256 }}
-  - url: https://github.com/tothuhien/lsd2/archive/refs/tags/v.2.4.1.tar.gz
```

Comment:
@martin-g the previous recipe pinned the version of LSD2 - which means there was a mismatch between the version of LSD used by iqtree2 and LSD2 - something that's quite bad.

The previous recipe is wrong in that it uses the wrong LSD version/commit: https://github.com/tothuhien/lsd2/compare/125720d9da1281642053713d9b094d5bff2ce9cf..3062cd1d0adc3bb07e5786b30be0da308ce5ac8d

These are the differences.

I don't know of a good way to keep the main and subrepo in sync other than git url. But happy to hear if there's a way. maybe @aliciaaevans or @mencian have an idea.

Reply:
Good question. I know the git url is not preferred, but I'm not sure what would be better. I'll ask around a bit.

Reply:
I can switch to normal URL, it's just going to cause version mismatch, I'm certain of that 🙈

Workaround could be to put warnings all over the recipe for reviewers, to check exact shas of submodules.